[ADD] save_discard_button module

synconics opened the pull request.

sam-synconics committed:
[ADD] save_discard_button module

[MOD] Index

[MOD] Menifest

[ADD] readme folder

[MOD] readme folder

[RID] unnecessary things

sam-synconics committed:
[ADD] save_discard_button module

sam-synconics committed:
[MOD] Index

sam-synconics committed:
[MOD] Menifest

legalsylvain commented:
Hi @synconics. 

1) To fix the pre-commit error, you should run pre-commit locally in your repo.
if not installed, ``pipx install pre-commit``
Then : ``pre-commit run -a`` in your local web folder.

2) The presentation of the module doesn't match with OCA rules. you should : 
- create a folder ``readme`` in the module
- add  ``rst`` file. at least mandatory DESCRIPTION.rst`` file, that describe the module. (you can make a reference to the image present in the description folder).
- see : template : https://github.com/OCA/maintainer-tools/tree/master/template/module/readme

regards.

sam-synconics committed:
[ADD] readme folder

sam-synconics committed:
[MOD] readme folder

synconics commented:
Hi @legalsylvain 

Thanks for your assistance. We have done all steps as per your suggestion.

legalsylvain commented on save_discard_button/readme/DESCRIPTION.rst:
```diff
@@ -0,0 +1,3 @@
+Change Save & Discard Button style.
+
+.. image:: ./static/description/icon.png
```
```suggestion
.. image:: ../static/description/save_button.png
```

2 minor fixes : 
- use a bigger image in the description file. (that will autogenerate index.html file and main README.rst file)
- use correct relative path to make the display working everywhere.

legalsylvain reviewed (commented):
Hi @synconics : thanks a for your time.

Some minor points regarding contributing in OCA : 

1) you can remove the following files that are unrelated to the current module : 
- save_discard_button/static/description/Odoo_Editions_All.png
- save_discard_button/static/description/gmail_icon.png
- save_discard_button/static/description/main_banner.png
- save_discard_button/static/description/main_screen.png
- save_discard_button/static/description/save_dis_icon.png
- save_discard_button/static/description/skype.png
- save_discard_button/static/description/test.png
- save_discard_button/static/description/whatsapp.png
- save_discard_button/static/description/whatsapp_barcode.png

2) you can also remove the following file that will be erased by the bot, when merging.
- save_discard_button/static/description/index.html

3) If you add a CONTRIBUTORS.rst file and / or a CREDITS.rst file (in the readme folder), you can add a reference to your company.
4) you can add ``"maintainers": ["synconics"]``, in the manifest, so that you will pinged if a people propose a PR that is changing / migrating your module.

5) Once done, you can squash all your commits.

Code review : LGTM. Not an expert in CSS, but looks correct. CC : @OCA/web-maintainers 

functional review : this module makes sense ! specially for people that was using odoo < 16. The two new icons introduced by Odoo doesn't makes sense, and writing explicitely the actions names is better in my opinion. :+1:

sam-synconics committed:
[RID] unnecessary things

synconics commented:
Hi @legalsylvain 

Thanks for your suggestion, I have updated accordingly.

legalsylvain reviewed (approved):
can you squash all the commits ? 

except that point :+1: thanks !

sam-synconics committed:
Merge branch '16.0' of https://github.com/synconics/web into 16.0

synconics commented:
Hi @legalsylvain 

I have executed squash command.

synconics commented:
Hi @legalsylvain 

Application is not publish yet, we need to do anything else from our side?

legalsylvain commented:
The oca workflow is : 
- propose a PR.
- answer the question / fix the error.
- wait for two approvals and 5 days. (just one approval for the time being).

Then the PR can be merged by a maintainer. (I am).
Once merged, the module is gradually becoming available on pypi, apps.odoo.com, https://odoo-community.org/shop... 

Regards.

Rad0van reviewed (approved):
Seems to work fine. The only thing is the naming of the module. Shouldn't it be named `web_save_discard_button` to be consistent with the rest of the repository? @legalsylvain  what's your opinion?

legalsylvain commented:
> Shouldn't it be named web_save_discard_button to be consistent with the rest of the repository?

:+1:

Rad0van reviewed (changes requested):
Please rename the module as suggested.

Tardo commented on save_discard_button/static/src/scss/indicator_button.scss:
```diff
@@ -0,0 +1,64 @@
+        .fa-cloud-upload {
+            width: 100%;
+            &:before {
+                content: "Save";
```
Is this translatable?

synconics replied:
This is not translatable, because in css not allowed to translate.

Tardo commented on save_discard_button/static/src/scss/indicator_button.scss:
```diff
@@ -0,0 +1,64 @@
+        .fa-undo {
+            width: 100%;
+            &:before {
+                content: "Discard";
```
Is this translatable?

synconics replied:
This is not translatable, because in css not allowed to translate.

Tardo commented on save_discard_button/static/src/scss/indicator_button.scss:
```diff
@@ -0,0 +1,64 @@
+                transition: none;
+                width: 100%;
+                text-transform: uppercase;
+                font-family: "ubuntu";
```
Please add more families...
Maybe you can use: `$o-system-fonts`

synconics replied:
Hi @Tardo I have updated.

Tardo commented on save_discard_button/static/src/scss/indicator_button.scss:
```diff
@@ -0,0 +1,64 @@
+                transition: none;
+                width: 100%;
+                text-transform: uppercase;
+                font-family: "ubuntu";
```
Please add more families...

synconics replied:
Hi @Tardo I have updated.

sam-synconics committed:
[MOD] Font

sam-synconics committed:
[Rename] Module

[Resolve] pre-commit

[ADD] Module translation

Added translation using Weblate (Dutch)

Translated using Weblate (Dutch)

Currently translated at 100.0% (4 of 4 strings)

Translation: web-16.0/web-16.0-web_advanced_search
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_advanced_search/nl/

Translated using Weblate (Dutch)

Currently translated at 100.0% (10 of 10 strings)

Translation: web-16.0/web-16.0-web_company_color
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_company_color/nl/

Translated using Weblate (Dutch)

Currently translated at 100.0% (5 of 5 strings)

Translation: web-16.0/web-16.0-web_chatter_position
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_chatter_position/nl/

[IMP] Terminology

move all addons to __unported__, set all addons as not installable

[PRT] Port 8.0 of web_ir_actions_act_window_page

[FIX] Unwellformed RST

[IMP] Shorten summary a bit for cosmetic reasons

[IMP] Description

[IMP] Remove empty line

Add OCA as author of OCA addons

In order to get visibility on https://www.odoo.com/apps the OCA board has
decided to add the OCA as author of all the addons maintained as part of the
association.

[UPD] prefix versions with 8.0

[MIG] Make modules uninstallable

[MIG] Rename manifest files

run onclose handler if we got one

[MIG][10.0] web_ir_actions_act_window_page

[ADD] Demo data for testing

OCA Transbot updated translations from Transifex

[UPD] Update web_ir_actions_act_window_page.pot

[13.0][MIG] web_ir_actions_act_window_page

[16.0][MIG] web_ir_actions_act_window_page

[UPD] Update web_ir_actions_act_window_page.pot

[UPD] addons table in README.md

[UPD] README.rst

[ADD] setup.py

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: web-16.0/web-16.0-web_ir_actions_act_window_page
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_ir_actions_act_window_page/

sam-synconics committed:
[Rename] Module

sam-synconics committed:
[Resolve] pre-commit

synconics commented:
@Rad0van Module name is updated as per your suggestion.

remi-filament commented:
Thanks @synconics that's a good idea, however if we could not find a way to translate the buttons, it is not usable for my use cases with my customers since everything else is translated...
Also you should squash all your commits in 1 commit before it could be approved
Best Regards

legalsylvain commented:
Hi @remi-filament and @Tardo : do you have a proposal to make that texts translatable ? 

AFAIK, such "content" directive are present in odoo core, and are not translated.

https://github.com/odoo/odoo/blob/16.0/addons/web/static/lib/tempusdominus/tempusdominus.scss#L122

Tardo commented:
@legalsylvain For this module, I think the 'web.FormStatusIndicator' template should be changed. 

** Also when using "qweb client" you have to add something like:
```python
class IrHttp(models.AbstractModel):
    _inherit = "ir.http"

    @classmethod
    def _get_translation_frontend_modules_name(cls):
        modules = super()._get_translation_frontend_modules_name()
        return modules + ["web_save_discard_button"]
```

sam-synconics committed:
[ADD] Module translation

sam-synconics committed:
Merge branch '16.0' of https://github.com/OCA/web into 16.0

sam-synconics committed:
Merge branch '16.0' of https://github.com/synconics/web into 16.0

sam-synconics committed:
[squash] commit

synconics commented:
@Tardo , I have updated code.

synconics commented:
@remi-filament I have executed squash commit command.

synconics commented:
Hi @remi-filament @Tardo 

Application is not publish yet, we need to do anything else from our side?

legalsylvain commented:
> Application is not publish yet, we need to do anything else from our side?

Answered here : https://github.com/OCA/web/pull/2577#issuecomment-1663886970

Note : your commits are not squashed.

Tardo commented:
@synconics thx for the changes... You still need to change the 'web.FormStatusIndicator' template instead of using the 'content' CSS attribute for the translations issue.

You can see an example here: https://github.com/OCA/web/blob/28540e66a46978ded16cf59a3463b559dc0d07c9/web_responsive/static/src/legacy/xml/form_buttons.xml#L74C5-L86C9

sam-synconics committed:
[ADD] Template

sam-synconics committed:
[Pre-commit]

synconics commented:
@Tardo I have updated code as per your suggestion.

synconics commented:
@legalsylvain When I am doing the squashed commit then module name is change as per old name "save_discard_button" instead of "web_save_discard_button".

legalsylvain commented:
you should run something like ``git rebase -i HEAD~18`` and squach all the commits.

synconics commented:
@legalsylvain I have executed which command you have mention but when I pick first commit and squash all other commits then module name is changed again as per previous "save_discard_button" name, but as per suggested by @Rad0van I have changed module name to "web_save_discard_button".
So, now how I can remain module name as per updated name in squash commit?

legalsylvain commented:
Sorry. I don't know how to help you. in the worst case, you can create a new branch and a new PR...

remi-filament commented:
Hi @synconics 
Rebase is rewinding to your first commit and then applying changes one by one, so it is normal that the first actions required refer to the old module name, once you will reach the commit where you changed names, it should be applied as well.
The problem here is that your history is a bit of a mess since you have merge commits and commits from other repos added in the meantime. I think the easiest way to fix would be recreating your PR (or at least your branch with only one commit !)
